include/zephyr/sys/uuid.h

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2024, SECO Mind Srl
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UUID_H_
+#define ZEPHYR_INCLUDE_SYS_UUID_H_
+
+/**
+ * @file
+ *
+ * @brief Utility functions for the generation and parsing of Universal Unique Identifier.
+ * @details This driver is compliant with RFC9562: https://datatracker.ietf.org/doc/rfc9562/
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup uuid UUID
+ * @since 4.0
+ * @version 0.1.0
+ * @ingroup utilities
+ * @{
+ */
+
+/** @brief Number of bytes in the binary representation of a UUID. */
+#define UUID_SIZE 16U
+
+/** @brief Length of the UUID canonical string representation, including the NULL terminator. */
+#define UUID_STR_LEN 37U
+
+/** @brief Length of the UUID base64 string representation, including the NULL terminator. */
+#define UUID_BASE64_LEN 25U
+
+/**
+ * @brief Length of the UUID base64 URL and filename safe string representation, including the
+ * NULL terminator.
+ */
+#define UUID_BASE64URL_LEN 23U
+
+/** @brief Binary representation of a UUID. */
+struct uuid {
+	/** @cond INTERNAL_HIDDEN */
+	uint8_t val[UUID_SIZE];
+	/** @endcond */
+};
+
+/**
+ * @brief Generate a UUIDv4.
+ *
+ * @param out The UUID where the result will be written.
+ *
+ * @retval 0 The UUID has been correctly generated and stored in @p out
+ * @retval -EINVAL @p out is not acceptable
+ */
+int uuid_generate_v4(struct uuid *out);
+
+/**
+ * @brief Generate a UUIDv5.
+ *
+ * @details This function computes a deterministic UUID starting from a namespace UUID and binary
+ * data.
+ *
+ * @param[in] namespace The string representation of an UUID to be used as namespace.
+ * @param[in] data A pointer to the data that will be hashed to produce the UUID.
+ * @param[in] data_size The size of the data buffer.
+ * @param[out] out The UUID where the result will be written.
+ *
+ * @retval 0 The UUID has been correctly generated and stored in @p out
+ * @retval -EINVAL @p out is not acceptable
+ * @retval -ENOMEM Memory allocation failed
+ * @retval -ENOTSUP mbedTLS returned an unrecognized error
+ */
+int uuid_generate_v5(struct uuid namespace, const void *data, size_t data_size, struct uuid *out);
+
+/**
+ * @brief Copy an UUID into another UUID.
+ *
+ * @param dst Destination for the copy.
+ * @param src Source for the copy.
+ *
+ * @retval 0 The UUID has been correctly copied in @p dst
+ * @retval -EINVAL @p dst is not acceptable
+ */
+int uuid_copy(struct uuid *dst, struct uuid src);
```

Comment:
In most functions you also have `(src, dst)` (e.g. `uuid_from_buffer(const uint8_t data[UUID_SIZE], struct uuid *out);` , but this functions does `(dst, src)`. 

Consider making them consistent